static/app/components/events/interfaces/spans/embeddedSpanTree.tsx

```diff
@@ -0,0 +1,120 @@
+import React from 'react';
+import styled from '@emotion/styled';
+
+import ErrorBoundary from 'sentry/components/errorBoundary';
+import space from 'sentry/styles/space';
+import {Event, EventTransaction, Organization} from 'sentry/types';
+import EventView from 'sentry/utils/discover/eventView';
+import GenericDiscoverQuery from 'sentry/utils/discover/genericDiscoverQuery';
+import QuickTraceQuery from 'sentry/utils/performance/quickTrace/quickTraceQuery';
+import useApi from 'sentry/utils/useApi';
+import {useLocation} from 'sentry/utils/useLocation';
+
+import TraceView from './traceView';
+import {FocusedSpanIDMap} from './types';
+import WaterfallModel from './waterfallModel';
+
+type Props = {
+  event: Event;
+  organization: Organization;
+  projectSlug: string;
+  focusedSpanIds?: FocusedSpanIDMap;
+};
+
+// This is a wrapper class that is intended to be used within Performance Issues
+export function EmbeddedSpanTree(props: Props) {
+  const {event, organization, projectSlug, focusedSpanIds} = props;
+  const api = useApi();
+  const location = useLocation();
+
+  const eventView = EventView.fromNewQueryWithLocation(
+    {
+      id: undefined,
+      version: 2,
+      name: '',
+      fields: ['transaction.duration'],
+      projects: [1],
+      query: 'event.type:transaction',
+      environment: [],
+    },
+    location
+  );
+
+  return (
+    <React.Fragment>
+      <ErrorBoundary mini>
+        <QuickTraceQuery event={event} location={location} orgSlug={organization.slug}>
+          {results => {
+            if (results.isLoading) {
+              return null;
```

Comment:
I think we need to handle loading here in a more explicit way like displaying loaders. Something we can add in a next PR because span tree is likely to stay as a perf-specific section in issue details

Reply:
That's true, I can quickly add this in right now

Reply:
nw, that can be a follow up clean up PR 👍 and some tests would be great too

Reply:
Sounds good1